src/main/groovy/com/cloudbees/groovy/cps/CpsTransformer.groovy

```diff
@@ -78,7 +79,7 @@ import static org.codehaus.groovy.syntax.Types.*
  * @author Kohsuke Kawaguchi
  */
 class CpsTransformer extends CompilationCustomizer implements GroovyCodeVisitor {
-    private int iota=0;
+    private static final AtomicLong iota = new AtomicLong();
```

Comment:
@jglick My bug-sense (like Spidey sense but nerdier) is tingling from that -- at the very least we're mandating synchronized access across all pipelines which is a potential multithreading bottleneck.